compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java

```diff
@@ -1000,6 +1001,33 @@ public void visit(BLangConstrainedType constrainedTypeNode) {
         resultType = constrainedType;
     }
 
+    private BType getXMLConstrainedType(BType constraintType, DiagnosticPos pos) {
+        BType constrainedType;
+        if (constraintType.tag != TypeTags.UNION) {
+            if (!TypeTags.isXMLTypeTag(constraintType.tag)) {
```

Comment:
What about if the constraint type happens to be a finite type, which contains a set of XML values?

Reply:
Can we pass `xml values` in a  finite type?  @rdhananjaya

Reply:
@pubudu91 could you provide an example for this?
  constraints here should be only union (1 member or up) of xml subtypes.

Reply:
I think what @pubudu91  meant is something like this.
```ballerina
type Foo "bar"|"car";
```
but for **xml-values**

Reply:
I don't see how it's possible in the current language, that's why I asked for an example.